floodviz/static/js/map.js

```diff
@@ -1,187 +1,321 @@
+				var gage_point_cords = document.getElementById('map' + sitekey).getBoundingClientRect();
+				maptip.transition().duration(500);
+				maptip.style("display", "inline-block")
+					.style("left", (gage_point_cords.left) + 7 + "px")
+					.style("top", (gage_point_cords.top - 30) + "px")
+					.html((sitename));
+			};
+			/**
+			 * Removes tooltip style from map site.
+			 */
+			self.site_tooltip_remove = function () {
+				maptip.style("display", "none");
+			};
+
+			/**
+			 * Remove/Add accent for a svg circle representing a site.
+			 * Used by hydromodule for cross figure interactions.
+			 */
+			self.site_remove_accent = function (sitekey) {
+				d3.select('#map' + sitekey).attr('class', 'gage-point');
+			};
+			self.site_add_accent = function (sitekey) {
+				d3.select('#map' + sitekey).attr('class', 'gage-point-accent');
+			};
+			self.click = function (sitekey) {
+				var new_display_ids = FV.hydrograph_display_ids;
+				var being_displayed = new_display_ids.indexOf(sitekey) !== -1;
+				if (being_displayed === true) {
+					self.site_remove_accent(sitekey);
+					new_display_ids.splice(new_display_ids.indexOf(sitekey), 1);
+					options.hover_out(sitekey);
+				}
+				else {
+					self.site_add_accent(sitekey);
+					new_display_ids.push(sitekey);
+				}
+				options.click_toggle(new_display_ids);
 				options.hover_in(sitekey);
-			}
-			options.click_toggle(new_display_ids);
-			options.hover_in(sitekey);
-		};
-		return self
-	};
+			};
+
+
+			var select_box_start = function (p) {
+				if (d3.event.defaultPrevented) return;
+				svg.append('rect')
+					.attr('x', p[0])
+					.attr('y', p[1])
+					.attr('height', 0)
+					.attr('width', 0)
+					.attr('class', 'select-box')
+					.attr('id', 'map-select-box');
+
+				state.box = {
+					x: p[0],
+					y: p[1],
+					height: 0,
+					width: 0
+				};
+			};
+
+			var select_box_drag = function (p) {
+				var box = d3.select('#map-select-box');
+				if (!box.empty()) {
+					var d = {
+						x: parseInt(box.attr("x")),
+						y: parseInt(box.attr("y")),
+						width: parseInt(box.attr("width")),
+						height: parseInt(box.attr("height"))
+					};
+					var move = {
+						x: p[0] - d.x,
+						y: p[1] - d.y
+					};
 
-}());
+					if (move.x < 1 || (move.x * 2 < d.width)) {
+						d.x = p[0];
+						d.width -= move.x;
+					}
+					else {
+						d.width = move.x;
+					}
+
+					if (move.y < 1 || (move.y * 2 < d.height)) {
+						d.y = p[1];
+						d.height -= move.y;
+					}
+					else {
+						d.height = move.y;
+					}
+
+					box.attr('x', d.x)
+						.attr('y', d.y)
+						.attr('width', d.width)
+						.attr('height', d.height);
+
+					state.box = {
+						x: d.x,
+						y: d.y,
+						height: d.height,
+						width: d.width
+					};
+				}
+			};
+
+			var select_box_end = function () {
+				// Check if the box has a reasonable area to make sure it isn't a click by mistake
+				var area = state.box.width * state.box.height;
+				if (area >= 100) {
```

Comment:
I don't think this is something that we're going to need to change often, and it's only used once, so I'm going to leave it as-is